src/object/srv_obj.c

```diff
@@ -1238,6 +1238,43 @@ daos_iod_recx_dup(daos_iod_t *iods, uint32_t iod_nr, daos_iod_t **iods_dup_ptr)
 	return 0;
 }
 
+static bool
+obj_fetch_from_snapshot(struct obj_rw_in *orw, struct obj_io_context *ioc)
+{
+	struct ds_cont_child	*cont;
+	int			 i;
+
+	cont = ioc->ioc_coc;
+	for (i = 0; i < cont->sc_snapshots_nr; i++) {
```

Comment:
Is there any other way to tell it's a snapshot fetch (like from fetch RPC)?

Snapshot IV might not be propagated yet when fetch request arrives (and snapshot list could be changed on snapshot create/delete), this approach looks not quite right to me.

Reply:
agreed, could define a new RPC ORF_FETCH_SNAP flag

"Snapshot IV might not be propagated yet when fetch request arrives " in that case the aggregation has not happen (aggregate will first check the snapshot list). So, if the list is empty, at that time need not worry about it?
But yes, the snapshot list can be changed.

Reply:
`So, if the list is empty, at that time need not worry about it?`

If the list is empty, a snapshot fetch will be mistakenly regard as normal fetch, right?